protocol/contracts/TempleZaps.sol

```diff
@@ -0,0 +1,190 @@
+pragma solidity ^0.8.4;
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./GenericZap.sol";
+
+
+interface ITempleStableRouter {
+  function tokenPair(address token) external view returns (address);
+  function swapExactStableForTemple(
+    uint amountIn,
+    uint amountOutMin,
+    address stable,
+    address to,
+    uint deadline
+  ) external returns (uint amountOut);
+  function swapExactTempleForStable(
+    uint amountIn,
+    uint amountOutMin,
+    address stable,
+    address to,
+    uint deadline
+  ) external returns (uint);
+  function addLiquidity(
+    uint amountADesired,
+    uint amountBDesired,
+    uint amountAMin,
+    uint amountBMin,
+    address stable,
+    address to,
+    uint deadline
+  ) external returns (uint amountA, uint amountB, uint liquidity);
+  function swapExactStableForTempleQuote(address pair, uint amountIn) external view returns (uint amountOut);
+  function swapExactTempleForStableQuote(address pair, uint amountIn) external view returns (bool priceBelowIV, uint amountOut);
+  function quote(uint amountA, uint reserveA, uint reserveB) external pure returns (uint amountB);
+}
+
+interface IVaultProxy {
+  function getFaithMultiplier(uint256 _amountFaith, uint256 _amountTemple) pure external returns (uint256);
+  function faithClaimEnabled() external view returns (bool);
+}
+
+interface IVault {
+  function depositFor(address _account, uint256 _amount) external; 
+}
+
+interface IFaith {
+  // User Faith total and usable balance
+  struct FaithBalance {
+    uint112 lifeTimeFaith;
+    uint112 usableFaith;
+  } 
+
+  function balances(address user) external view returns (FaithBalance memory);
+  function gain(address to, uint112 amount) external;
+  function redeem(address to, uint112 amount) external;
+}
+
+interface ILockedOGTemple {
+  function OG_TEMPLE() external ;
+  function withdrawFor(address _staker, uint256 _idx) external; 
+}
+
+interface IGenericZaps {
+  function zapIn(
+    address fromToken,
+    uint256 fromAmount,
+    address toToken,
+    uint256 amountOutMin,
+    address swapTarget,
+    bytes calldata swapData
+  ) external returns (uint256 amountOut);
+}
+
+
+contract TempleZaps is Ownable {
```

Comment:
Is this strictly required, or really just a helper so there's only one approval step?

Would the idea be to implement an equivalent for Stax?

Reply:
temple has specific requirements to zapping into vaults, hence separation of concerns with a dedicated contract which utilizes functions from generic zaps.
yes, stax would have its own after we specify requirements and there are unique ones. so far, stax can use everything in the generic zaps